numpy/core/tests/test_scalarmath.py

```diff
@@ -678,11 +678,35 @@ def _test_abs_func(self, absfunc, test_dtype):
         x = test_dtype(np.finfo(test_dtype).min)
         assert_equal(absfunc(x), -x.real)
 
-    @pytest.mark.parametrize("dtype", floating_types + complex_floating_types)
+    @pytest.mark.parametrize(
+        "dtype",
+        [
+            pytest.param(
+                dtype,
+                marks=pytest.mark.xfail(
+                    sys.platform == "cygwin" and dtype == np.clongdouble,
+                    reason="npy_cabsl calls npy_hypotl, which is npy_hypot",
+                ),
+            )
+            for dtype in floating_types + complex_floating_types
+        ],
+    )
     def test_builtin_abs(self, dtype):
```

Comment:
A cleaner way to do this is to raise skip inside the function. Something like
```
def test_builtin_abs(self, dtype):
    if dtype == np.clongdouble and sys.platform == "cygwin" :
        pytest.skip("absl is computed in double precision on cygwin")
    ...
```
You could also use `pytest.xfail` instead of `pytest.skip`.

Reply:
Done, here and for the next test.